xlators/nfs/server/src/nfs3-helpers.c

```diff
@@ -3693,8 +3693,8 @@ nfs3_fh_resolve_inode_lookup_cbk (call_frame_t *frame, void *cookie,
                 nfs3_call_resume (cs);
                 goto err;
         }
-
-        memcpy (&cs->stbuf, buf, sizeof(*buf));
+        if (!cs->setattr_valid)
+            memcpy (&cs->stbuf, buf, sizeof(*buf));
```

Comment:
This might be an optimization, saving memory for all cs objects.

Patches need to be sent through Gerrit, see https://gluster.readthedocs.io/en/latest/Developer-guide/Simplified-Development-Workflow for a description.